Initiative from Combat Tracker

KristjanLaane opened the pull request:
Initiative integration when rolled directly from the Combat Tracker. Option to also show number of actions available based on how many triggers were rolled - use fighting.actionsCurrent for the Tracked Resource under Combat Tracker Settings.

![CombatTrackerInitiative](https://user-images.githubusercontent.com/1708294/99715493-9e651480-2aaf-11eb-8328-1c1ea495811b.jpg)

KristjanLaane committed:
Initiative integration when rolled directly from the Combat Tracker. Option to also show number of actions available based on how many triggers were rolled - use fighting.actionsCurrent for the Tracked Resource under Combat Tracker Settings.

moo-man commented:
Isn't there a process by which you spend a certain amount of ego?

KristjanLaane commented:
Yes, spending ego points is not implemented here, but it's optional to spend ego points and can only be done on the first round. The use case here is if you have a number of characters in combat and you want to quickly roll all their initiatives each round - you can click Roll All and the rest is magic. I implemented it because I need it as I don't want to slow the fights down by individually rolling lots of initiatives, and do that again individually for each character involved every single round :)

moo-man commented:
Sure, I understand, just considering how to handle the ego spending in the future.

moo-man committed:
Added sync functions to improve initiative chat card display

moo-man commented:
my only issue here is the very basic chat card this produces.
![image](https://user-images.githubusercontent.com/28637157/99748697-a074a680-2aa2-11eb-8133-21fb90a1cbd2.png)

I've added synchronous roll functions so that initiative rolls can go through the same rendering process as normal rolls. I'm not very happy with this solution as it's duplicated code and redundant chat cards (the basic chat card is still displayed). But I'm open to thoughts.

moo-man committed:
Added spending ego on initiative roll (+D bonus not implemented yet)

KristjanLaane committed:
NB! Due to template changes, most of the following does not work with characters created before this commit - create new actors to use this stuff.

Native initiative chat message removed and replaced by bespoke Degenesis initiative chat message.
Initiative calculation takes spent ego into account now - successes from the extra dice, but not triggers.
Initiative from the Combat tab in Actor sheet now triggers the fully fledged initiative response as from the Combat Tracker.
Combat Tracker tracked resource changed to state.initiative.actions.

KristjanLaane commented:
After some digging found a way to remove the basic chat card and replaced it completely with a Degenesis-style one. 

Other things, too:
Initiative calculation rolls spent ego dice as well now - successes from the extra dice, but not triggers.
Initiative from the Combat tab in Actor sheet now triggers the fully-fledged initiative response, same as from the Combat Tracker.

Please note that due to template changes, most of this does not work with characters created before this commit - create new actors to use this stuff. 

Also, Combat Tracker tracked resource changed to state.initiative.actions - to get that number of actions to show as the secondary value in the Combat Tracker.

moo-man commented:
do you have a discord? I'd like to talk about this more.

moo-man committed:
Added spent ego bonus, spent ego counts for triggers.

moo-man commented on module/combat-degenesis.js:
```diff
@@ -51,41 +57,38 @@ export class DegenesisCombat extends Combat {
+            actionNumber: actor.data.data.attributes[actor.data.data.skills[skill].attribute].value + actor.data.data.skills[skill].value + spentEgo
         };
-        const rollResults = DegenesisDice.rollWithout3dDice(rollData);
+        const {rollResults, cardData} = actor.rollFightRollSync("initiative");
```
Since I assumed you removed this due to spent ego triggers not counting, I added it back since we determined that is not the case.

moo-man commented on module/actor/actor-degenesis.js:
```diff
@@ -463,12 +463,14 @@ export class DegenesisActor extends Actor {
         let cardData = this.constructCardData("systems/degenesis/templates/chat/roll-card.html", DEGENESIS.skills[skill])
 
+        if (type == "initiative")
+            cardData = this.constructCardData("systems/degenesis/templates/chat/initiative-roll-card.html", DEGENESIS.skills[skill] + " - " + "Initiative") 
```
I figured there may be ways to use different skills for initiative (I could be wrong), so showing the skill used seems useful.

moo-man commented:
Going to go ahead and merge this, feel free to make further changes in another PR if you want.
